src/main/java/com/target/studentinfo/service/StudentService.java

```diff
@@ -0,0 +1,71 @@
+package com.target.studentinfo.service;
+
+import com.target.studentinfo.dto.mapper.StudentMapper;
+import com.target.studentinfo.dto.request.StudentRequest;
+import com.target.studentinfo.dto.response.StudentResponse;
+import com.target.studentinfo.model.Student;
+import com.target.studentinfo.respository.StudentRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+
+public class StudentService {
+    private final StudentRepository studentRepository;
+
+    public StudentService(StudentRepository studentRepository) {
+        this.studentRepository = studentRepository;
+    }
+
+    public List<Student> getAllStudentInfo(){
+
+        return studentRepository.findAll();
+    }
+
+    public Optional<Student> getStudentById(UUID id){
+        return studentRepository.findById(id);
+    }
+
+    public Student addNewStudent(Student student){
+        return studentRepository.save(student);
+    }
+
+    public Student updateStudent(Student student) {
+        return studentRepository.save(student);
+    };
+
+    public void deleteStudent(UUID id) {
+        studentRepository.deleteById(id);
+    };
+
+    public StudentResponse addNewStudent(StudentRequest studentRequest) {
+
+        Student student = StudentMapper.MAPPER.fromReqToModel(studentRequest);
+        studentRepository.save(student);
+        return StudentMapper.MAPPER.fromModelToResponse(student);
+    };
+    public StudentResponse updateStudent(StudentRequest studentRequest,UUID id) {
+
+        Optional<Student> checkExists = getStudentById(id);
+        if (! checkExists.isPresent())
+            throw new RuntimeException("Employee ID" + id + " not found");
+        Student student = StudentMapper.MAPPER.fromReqToModel(studentRequest);
+        student.setId(id);
+        studentRepository.save(student);
+        return StudentMapper.MAPPER.fromModelToResponse(student);
+    };
+
+
+
+
+
+
```

Comment:
As discussed, we only need these functions in repo layer.

1. getStudents() -> To get all students including soft deleted ones.
2. getStudentsActive() -> To get only the active students.
3. getStudentById()
4. getStudentByIdActive()
5. addStudent()
6. updateStudent()
7. deleteStudent()

Please rename the functions to be this way. Don't need any additional functions other than these.